File: 第7章 曲线之美/7.4 外摆线/外摆线.cpp
/*
程序：外摆钱 
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
*/
// 绘制外摆线
void draw(float n, float end)
{
	const float PI = 3.14159;
    float a = 45;
	float b = a / n;
    float t = 0;
    while (t <= end) {
		float rad = PI / 180 * t;
        float x = (a + b) * cos(rad) - b * cos((a + b) / b * rad);
        float y = (a + b) * sin(rad) - b * sin((a + b) / b * rad);
        pen.line(x, y);
        pen.down();
        t += 1;
    }
}

int main()
{
	// 画笔初始化
	pen.color(_green4);
	pen.size(2);
	pen.up();
    draw(5.0/2, 2 * 360);
    return 0;
}

File: 第7章 曲线之美/7.4 外摆线/外摆线.py
'''
程序：外摆线
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
'''
from turtle import *
from math import pi, sin, cos

def draw(n, end):
    '''绘制外摆线'''
    a = 45
    b = a / n
    t = 0
    while t <= end:
        rad = pi / 180 * t
        x = (a + b) * cos(rad) - b * cos((a + b) / b * rad)
        y = (a + b) * sin(rad) - b * sin((a + b) / b * rad)
        goto(x, y)
        down()
        t += 1

if __name__ == '__main__':
    '''画笔初始化'''
    speed(0)
    pencolor('seagreen')
    pensize(2)
    up()
    draw(5/2, 2 * 360)
